samplers/derived.go

```diff
@@ -0,0 +1,11 @@
+package samplers
+
+import (
+	"github.com/stripe/veneur/ssf"
+)
+
+// DerivedMetricsProcessor processes any metric created from events or service checks into
+// the worker channels for flushing
+type DerivedMetricsProcessor interface {
```

Comment:
I think it's good to have 2 different interfaces since `signalfx` can just depend on the `DerivedMetricsProcessor` instead of the full on (metrics sink + derived metrics processor) super set interface. 

@asf-stripe you had feels about ^

Reply:
Yeah, I realized we need this interface definition locally within the package as well, so this is the right call.